climada_petals/hazard/copernicus_interface/index_definitions.py

```diff
@@ -0,0 +1,220 @@
+
+This module, `index_definitions.py`, defines and organizes the specifications, explanations, and variable mappings
+for various climate indices used within the CLIMADA seasonal forecast workflow. It centralizes key parameters,
+descriptions, filenames, and variable handling, ensuring consistent and extensible use across climate risk
+assessment workflows based on seasonal forecasts.
+
+Key features include:
+- Index Definitions: The `IndexSpec` dataclass defines attributes for each climate index, such as units,
+  standard and short names, full descriptions, required variables, and file naming conventions.
+- Enumerated Indices: The `IndexSpecEnum` organizes supported climate indices, such as Mean Temperature (Tmean),
+  Tropical Nights (TR), and Heat Wave (HW), each mapped to an `IndexSpec` instance for easy retrieval.
+- Variable-to-Short-Name Mapping: The `get_short_name_from_variable` function provides a centralized mechanism
+  for mapping variable names (e.g., "2m_temperature") to their corresponding short names (e.g., "t2m"), ensuring
+  consistency in variable handling.
+
+Attributes
+----------
+IndexSpec : dataclass
+    Contains index-specific attributes, including units, standard and short names, full names, filenames,
+    explanations, and variable requirements.
+
+IndexSpecEnum : Enum
+    Enumerates the supported climate indices, each mapped to its `IndexSpec` configuration.
+    
+Functions
+---------
+get_info(index_name)
+    Retrieves the complete specifications of a specified climate index, including its attributes and required variables.
+get_short_name_from_variable(variable)
+    Maps a variable's standard name (e.g., "2m_temperature") to its short name (e.g., "t2m").
+
+Example Usage
+-------------
+>>> from index_definitions import IndexSpecEnum, get_short_name_from_variable
+>>> index_info = IndexSpecEnum.get_info("TR")
+>>> print(index_info.explanation)
+Tropical Nights: Counts nights with minimum temperatures above a certain threshold. Default threshold is 20°C.
+>>> print(index_info.variables)
+['2m_temperature']
+
+>>> short_name = get_short_name_from_variable("2m_temperature")
+>>> print(short_name)
+t2m
+
+Notes
+-----
+This module relies on the `xarray`, `pandas`, and `logging` libraries for array manipulation, datetime handling,
+and error and information logging. It supports both index-based and variable-based workflows, allowing flexibility
+in how indices and variables are processed and managed.
+"""
+
+from enum import Enum
+from dataclasses import dataclass
+
+
+@dataclass
+class IndexSpec:
+    unit: str
+    full_name: str
+    explanation: str
+    variables: list
+
+
+class IndexSpecEnum(Enum):
+    HIA = IndexSpec(
+        unit="C",
+        full_name="Heat Index Adjusted",
+        explanation="Heat Index Adjusted: This indicator measures apparent temperature, considering both air temperature and humidity, providing a more accurate perception of how hot it feels.",
```

Comment:
not valid for extreme temperature (>=50 °C) and relative humidity conditions beyond the range the equation was developed for.